practice_coding_the_rails_way/sort_records_alphabetically.rb

```diff
@@ -1,4 +1,23 @@
 # 20. Sort Records Alphabetically
 
 # Make an array of hashes, each with a "title" (like blog posts).
-# Sort them alphabetically by title and print them.
\ No newline at end of file
+# Sort them alphabetically by title and print them.
+
+
+first_book = {:title => "To Kill a Mockingbird", :author => "Harper Lee"}
+second_book = {:title => "1984", :author => "George Orwell"}
+third_book = {:title => "The Great Gatsby", :author => "F. Scott Fitzgerald"}
+
+books = [first_book, second_book, third_book]
+sorted_books = books.sort_by { |book| book[:title] }
+
+print "Books Listed: \n"
+books.each do |book|
+    print "#{book[:title]} by #{book[:author]}\n"
+end
+
+print "\nHere are these books sorted alphabetically by Title: \n\n"
+
+sorted_books.each do |book|
+    print "#{book[:title]} by #{book[:author]}\n"
```

Comment:
line 16 and 21 are exactly the same. when you have things like this that are the same then you need to DRY it up (Don't Repeat Yourself). you can do it this way:

```
def display(book)
   puts "#{book[:title]} by #{book[:author]}"
end
```

and then you have

```
sorted_books.each do |book|
   display(book)
end
```